### PR TITLE
server/cluster: automatic gc tombstone store

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/5348

### What is changed and how does it work?

In a long-running cluster, there are too many tombstones. introduce auto gc mechanism.

```commit-message
server/cluster: automatic gc tombstone store
```

### Check List

Tests

- Unit test

```release-note
cluster: automatic gc tombstone store
```

### CODE REVIEW

**Comment by nolouch** [comment]
ptal @rleungx @HunDunDM

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -1523,6 +1524,20 @@ func (c *RaftCluster) checkStores() {
 	for _, store := range stores {
 		// the store has already been tombstone
 		if store.IsRemoved() {
+			if store.DownTime() > gcTombstoreInterval {
+				// TODO: remove this lock
```
Comment (bufferflies) [inline server/cluster/cluster.go:1528]: The lock can be ignore , the delete store locked has two step: 1. delete meta from etcd by txn and delete store from store map(have store lock).

**Review by lhy1024 — COMMENTED** [review]
Why use lock? the rest LGTM

**Review by rleungx — COMMENTED** [review]
LGTM, except for the comment.

File: server/cluster/cluster.go (hunk 1 above)
Comment (nolouch) [reply server/cluster/cluster.go:1528]: Because the function name warns me to use the lock.

**Comment by nolouch** [comment]
PTAL @bufferflies @rleungx

**Review by lhy1024 — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**